### PR TITLE
[Debt] Removes outdated comments

### LINKED ISSUE #11337: ♻️ Outdated `PoolTable.tsx` TODO comments
## ♻️ Debt

> All of these comments...can be removed since they are now present in the filters

https://github.com/GCTC-NTGC/gc-digital-talent/discussions/11190#discussioncomment-10389171

## 🕵️ Details

https://github.com/GCTC-NTGC/gc-digital-talent/blob/e97c51025ee277da255db8584f1bceaf231f1054/apps/web/src/pages/Pools/IndexPoolPage/components/PoolTable.tsx#L303

https://github.com/GCTC-NTGC/gc-digital-talent/blob/e97c51025ee277da255db8584f1bceaf231f1054/apps/web/src/pages/Pools/IndexPoolPage/components/PoolTable.tsx#L312

https://github.com/GCTC-NTGC/gc-digital-talent/blob/e97c51025ee277da255db8584f1bceaf231f1054/apps/web/src/pages/Pools/IndexPoolPage/components/PoolTable.tsx#L333

## 🙋‍♀️ Proposed Solution

Remove listed comments.

## ✅ Acceptance Criteria

- [ ] listed comments are removed from codebase

### PR DESCRIPTION
**Pull request by mnigh**
🤖 Resolves #11337.

## 👋 Introduction

This PR removes outdated comments.

## 🧪 Testing

1. Verify app and tests run as they did before

### CODE REVIEW

File: apps/web/src/pages/Pools/IndexPoolPage/components/PoolTable.tsx [hunk 1]
```diff
@@ -330,9 +328,8 @@ const PoolTable = ({ title, initialFilterInput }: PoolTableProps) => {
       ({ status }) => getLocalizedName(status?.label, intl),
       {
         id: "status",
-        // TO DO: Reenable when scope is added
         enableColumnFilter: false,
-        // TO DO: Reenable when relation order by added
+        // TODO: Re-enable when relation order by added.
         enableSorting: false,
```
Comment (vd1992) [inline apps/web/src/pages/Pools/IndexPoolPage/components/PoolTable.tsx:333]: Is there an issue up to do this?

Comment (mnigh) [reply apps/web/src/pages/Pools/IndexPoolPage/components/PoolTable.tsx:333]: Good question, you are not the first to ask 😆.

https://github.com/GCTC-NTGC/gc-digital-talent/discussions/11190#discussioncomment-10421081

Comment (vd1992) [reply apps/web/src/pages/Pools/IndexPoolPage/components/PoolTable.tsx:333]: Ahaha
Yeah, TODOs without an issue or someway to search them will likely never get actioned

Comment (mnigh) [reply apps/web/src/pages/Pools/IndexPoolPage/components/PoolTable.tsx]: [a68964b](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11339/commits/a68964be7a4350c91e971cef6a7ead7dc6a7c879)

**Review by vd1992 — APPROVED** [review]

**Pull request merged**